tesseract-server/src/handlers/logic_layer/geoadjacents.rs

```diff
@@ -108,24 +104,20 @@ pub fn logic_layer_geoadjacent(
         None => return Ok(HttpResponse::NotFound().json("Unable to access cube cache".to_string()))
     };
 
-    let cuts_map = match agg_query.cuts{
+    let cuts_map = match agg_query.cuts {
         Some(cut_map) => cut_map,
-        None => HashMap::new(),
+        None => return Ok(HttpResponse::NotFound().json("Malformed cuts".to_string())),
     };
 
     let level_map = &cube_cache.level_map;
     let property_map = &cube_cache.property_map;
     let geoservice_url = &req.state().env_vars.geoservice_url;
 
-    let dimensions_map = match getdimensions(&cuts_map, &cube, &cube_cache, &level_map, &property_map, &geoservice_url) {
+    let dimensions_map: Vec<Vec<String>> = match get_relations(&cuts_map, &cube, &cube_cache, &level_map, &property_map, &geoservice_url) {
         Ok(dm) => dm,
-        Err(err) => return Ok(HttpResponse::NotFound().json(err.to_string()))
+        Err(err) => Vec::new(),
```

Comment:
We should be returning any errors that come from the `get_relations` method here instead of initializing an empty vector. If your goal here is to have a way to return an empty array when there are no matches, you should return that explicitly from the `get_relations` function. That way we are still returning any possible errors.